lib_xua/doc/rst/sw_dfu.rst

```diff
@@ -3,25 +3,117 @@
+
+The DFU interface is enabled by default (See XUA_DFU_EN define in ``xua_conf_default.h``).
+When DFU is enabled, there are two sets of descriptors that the device can export, depending on the mode in which it operates.
+There's a descriptor set for the runtime mode, which is the mode the device normally operates in and a set of descriptors for the DFU mode.
+
+In the runtime mode, the DFU interface is one of potentially multiple interfaces that the device exposes. :ref:`dfu_interface_runtime` shows the DFU interface
+descriptors when enumerating in runtime mode as seen in a `Beagle USB analyser <https://www.totalphase.com/products/data-center/>`_ trace:
+
+ .. _dfu_interface_runtime:
+
+ .. figure:: images/dfu_interface_runtime_mode.png
+   :width: 100%
+
+   DFU interface when part of runtime mode descriptor set
+
+Note the **bInterfaceProtocol** field set to **Runtime**.
+
+In DFU mode, the device exports the DFU descriptor set. The DFU mode descriptors specify only one interface, the DFU interface.
+:ref:`dfu_interface_dfu` shows the DFU interface
+descriptors when enumerating in DFU mode as seen in a Beagle USB analyser trace:
+
+ .. _dfu_interface_dfu:
+
+ .. figure:: images/dfu_interface_dfu_mode.png
+   :width: 100%
+
+   DFU interface when part of DFU mode descriptor set
+
+Note the **bInterfaceProtocol** field set to **DFU mode**.
+
+Before starting the DFU upload or download process, the host sends a ``DFU_DETACH`` command to detach the device from runtime to DFU mode.
+In response to the ``DFU_DETACH`` command, the device reboots itself in DFU mode and enumerates using the DFU mode descriptors.
+Once the device is in DFU mode, the DFU interface can accept commands defined by the
+`DFU 1.1 class specification <http://www.usb.org/developers/devclass_docs/DFU_1.1.pdf*USB>`_.
+
+After detaching the device, the host proceeds with the DFU download/upload commands to write/read the firmware upgrade image to/from the device.
+Once the DFU download or upload process is complete, the host sends a ``DETACH`` command, and the device reboots itself back in runtime mode.
+
+.. note::
+
+   It is recommended that the runtime mode and DFU mode descriptors have different product IDs. This is to ensure that the host operating
+   system loads the correct driver as the device switches between runtime and DFU modes. The runtime and DFU PID are defined as overridable
+   defines ``PID_AUDIO_2`` and ``DFU_PID`` respectively in ``xua_conf_default.h``. Users can define custom PIDs in their application by overriding these defines.
+
+
+During the DFU download process, on receiving the first DFU_DNLOAD command (``wBlockNum`` = 0), the device erases the entire
+``FLASH_MAX_UPGRADE_SIZE`` bytes of the upgrade section of the flash. This is done by repeatedly calling ``flash_cmd_start_write_image``
+and can take several seconds. To avoid stalling the USB bus and risking a timeout, the flash erase is instead done in the DFU_GETSTATUS handling
```

Comment:
```suggestion
and can take several seconds. To avoid stalling the USB bus and risking a timeout, the flash erase is instead done in the ``DFU_GETSTATUS`` handling
```